src/rebar_prv_dialyzer.erl

```diff
@@ -539,6 +546,10 @@ get_config(State, Key, Default) ->
     Config = rebar_state:get(State, dialyzer, []),
     proplists:get_value(Key, Config, Default).
 
+debug_info(State) ->
+    Config = rebar_state:get(State, erl_opts, []),
+    proplists:get_value(debug_info, Config, false).
```

Comment:
Should I be trying to use `rebar_opts:erl_opts/1` here? I tried this:

```erlang
Config = rebar_opts:erl_opts(State),
proplists:get_value(debug_info, Config, false).
```

But it crashes:

```
$ DEBUG=1 ~/src/github.com/jkakar/rebar3/rebar3 dialyzer
===> Expanded command sequence to be run: [{default,app_discovery},
                                                  {default,install_deps},
                                                  {default,lock},
                                                  {default,compile},
                                                  {default,dialyzer}]
===> Provider: {default,app_discovery}
===> Provider: {default,install_deps}
===> Verifying dependencies...
===> Provider: {default,lock}
===> Provider: {default,compile}
===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", pre_hooks, compile) -> no hooks defined

===> Compiling rf
===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", pre_hooks, compile) -> no hooks defined

===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", pre_hooks, erlc_compile) -> no hooks defined

===> erlopts []
===> files to compile ["/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rf.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_error.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_typecheck_binary_op.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_compile_erlang.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_parse.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_scan.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_annotate_locals.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_compile.erl",
                              "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/src/rufus_typecheck_func_return_type.erl"]
===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", post_hooks, erlc_compile) -> no hooks defined

===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", pre_hooks, app_compile) -> no hooks defined

===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", post_hooks, app_compile) -> no hooks defined

===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", post_hooks, compile) -> no hooks defined

===> run_hooks("/Users/jkakar/src/github.com/rufus-lang/rufus/rf", post_hooks, compile) -> no hooks defined

===> Provider: {default,dialyzer}
===> Dialyzer starting, this may take a while...
===> Updating plt...
===> Resolving files...
===> kernel modules: [user_sup,logger_olp,local_udp,inet_tcp,inet6_tcp,
                             group,global_group,gen_tcp,gen_sctp,disk_log_sup,
                             disk_log_1,wrap_log_reader,user,standard_error,
                             pg2,logger_server,logger_handler_watcher,
                             logger_formatter,logger,inet_db,
                             hipe_unified_loader,file_server,file_io_server,
                             error_logger,error_handler,erl_signal_handler,
                             erl_boot_server,disk_log_server,code_server,
                             application_starter,application_master,
                             application_controller,rpc,os,logger_filters,
                             kernel_refc,inet_res,inet_hosts,inet_dns,dist_ac,
                             raw_file_io_list,raw_file_io_delayed,
                             raw_file_io_compressed,logger_backend,
                             inet_tcp_dist,inet6_tcp_dist,inet,heart,erl_epmd,
                             seq_trace,raw_file_io_inflate,
                             raw_file_io_deflate,ram_file,inet_parse,
                             inet_gethost_native,file,code,user_drv,
                             raw_file_io_raw,logger_config,kernel_config,
                             inet_config,erts_debug,disk_log,logger_std_h,
                             logger_simple_h,logger_disk_log_h,global_search,
                             auth,logger_proxy,group_history,erl_reply,
                             net_kernel,kernel,global,erl_ddll,dist_util,
                             net_adm,logger_h_common,erl_distribution,
                             application,raw_file_io,logger_sup,local_tcp,
                             inet_udp,inet_sctp,inet6_udp,inet6_sctp,gen_udp]
===> stdlib modules: [zip,dets_sup,timer,supervisor,proc_lib,io_lib,
                             gen_server,filelib,file_sorter,erl_tar,erl_error,
                             dets_server,calendar,beam_lib,sys,sofs,sets,qlc,
                             proplists,ordsets,maps,lists,gb_trees,gb_sets,
                             eval_bits,ets,erl_expand_records,erl_bits,
                             digraph_utils,dets_utils,dets,c,shell_default,
                             rand,qlc_pt,orddict,io_lib_fread,io_lib_format,
                             gen_event,escript,erl_lint,edlin_expand,dict,
                             base64,unicode,supervisor_bridge,slave,re,queue,
                             filename,erl_parse,erl_compile,erl_abstract_code,
                             win32reg,uri_string,string,erl_posix_msg,math,
                             log_mf_h,error_logger_tty_h,error_logger_file_h,
                             digraph,io_lib_pretty,dets_v9,binary,array,
                             unicode_util,shell,pool,otp_internal,
                             erl_internal,erl_eval,random,ms_transform,
                             gen_statem,gen_fsm,gen,erl_scan,edlin,io,
                             erl_anno,erl_pp,epp]
===> rf modules: [rufus_typecheck_binary_op,rufus_error,
                         rufus_annotate_locals,
                         rufus_typecheck_func_return_type,rufus_parse,
                         rufus_compile,rf,rufus_compile_erlang,rufus_scan]
===> Checking 181 files in "/Users/jkakar/src/github.com/rufus-lang/rufus/rf/_build/default/rebar3_22.0.1_plt"...
===> Uncaught error in rebar_core. Run with DEBUG=1 to see stacktrace or consult rebar3.crashdump
===> Uncaught error: {badrecord,dict}
===> Stack trace to the error location:
[{dict,get_slot,2,[{file,"dict.erl"},{line,413}]},
 {dict,find,2,[{file,"dict.erl"},{line,142}]},
 {rebar_opts,get,3,
             [{file,"/Users/jkakar/src/github.com/jkakar/rebar3/src/rebar_opts.erl"},
              {line,21}]},
 {rebar_opts,erl_opts,1,
             [{file,"/Users/jkakar/src/github.com/jkakar/rebar3/src/rebar_opts.erl"},
              {line,34}]},
 {rebar_prv_dialyzer,debug_info,1,
                     [{file,"/Users/jkakar/src/github.com/jkakar/rebar3/src/rebar_prv_dialyzer.erl"},
                      {line,550}]},
 {rebar_prv_dialyzer,run_dialyzer,3,
                     [{file,"/Users/jkakar/src/github.com/jkakar/rebar3/src/rebar_prv_dialyzer.erl"},
                      {line,477}]},
 {rebar_prv_dialyzer,check_plt,5,
                     [{file,"/Users/jkakar/src/github.com/jkakar/rebar3/src/rebar_prv_dialyzer.erl"},
                      {line,347}]},
 {rebar_prv_dialyzer,do,3,
                     [{file,"/Users/jkakar/src/github.com/jkakar/rebar3/src/rebar_prv_dialyzer.erl"},
                      {line,158}]}]
===> When submitting a bug report, please include the output of `rebar3 report "your command"`
```

Reply:
you actually want to look for `debug_info` as a boolean but there are other options as well supported by the compiler, including:
- `debug_info_key` 
- `encrypt_debug_info` 
- `{debug_info, {Backend, Data}}` (for other languages)

Those all mean debug info is possibly used.